### ISSUE TITLE
Custom behavior

### ISSUE THREAD

**Issue by ptcc** [mautic/mautic#742]
Hi, I would like to have some advice on how to implement some custom behavior:
What I would like to do is to update a bunch of lead fields when one specific field is updated.
I would have a table with the "trigger" field value and the corresponding values for the fields I also want to update.

The purpose is to change some "variables" that later i'm using in emails depending on the lead's history on my website.

I think there is no out of the box way of doing it, right?

I think I will have to hook something inside PageModel.php in hitPage function.

any thoughts?
thanks

**Comment by alanhartless** [comment]
It sounds like you'll need a custom plugin that listens to the lead edit event then handle the update logic or use a combination of 1.2.0's webhook to listen for a lead update then use the REST API to update the lead.  

I was thinking that it may be possible to leverage the foundation being added with #906 as I'm sure with it we'll be adding a lead profile condition.  But the campaign would only apply once and not ongoing so not sure if that would work for your scenario or not.  

Bottom line is I don't think there is a means to do this right now without some extra work and custom coding.
